apps/aechannel/src/aesc_utils.erl

```diff
@@ -341,9 +442,229 @@ process_solo_close_slash(ChannelId, FromPubKey, Nonce, Fee,
             {ok, last_onchain} ->
                 aesc_channels:close_solo(Channel0, PoI, Height)
         end,
-    ChannelsTree1 = aesc_state_tree:enter(Channel1, ChannelsTree0),
-
-    Trees1 = aec_trees:set_accounts(Trees, AccountsTree1),
-    Trees2 = aec_trees:set_channels(Trees1, ChannelsTree1),
+    Trees2 = set_channel(Channel1, Trees1),
     {ok, Trees2}.
 
+process_force_progress(ChannelId, FromPubKey, Nonce, Fee,
+                       _Payload, SoloPayload, Addresses,
+                       PoI, TxHash, Height, Trees) ->
+    {ok, [Channel, {_SoloSignedState, SoloPayloadTx}]} =
+          get_vals([get_channel(ChannelId, Trees),
+                   deserialize_payload(SoloPayload)]),
+    [Update] = aesc_offchain_tx:updates(SoloPayloadTx),
+    {ContractPubkey, Caller} = aesc_offchain_update:extract_call(Update),
+    %% use in gas payment
+    PoITrees0 = trees_from_poi(Addresses, PoI),
+    Reserve = aesc_channels:channel_reserve(Channel),
+    NextRound = aesc_offchain_tx:round(SoloPayloadTx),
+    PoITrees =
+        try aesc_offchain_update:apply_on_trees(Update, PoITrees0, NextRound,
+                                                      Reserve) of
+            UpdatedPoITrees -> UpdatedPoITrees
+        catch _:_ ->
```

Comment:
If anything goes wrong with the contract call updating the state tree provided - we go here. Then a new contract call is created (with an `error` return type and `invalid_call` as a value) consuming all gas provided.

This might happen if the contract call refers to another address (either a contract or an account) that is not present in the trees for example. Since we are not running the contract in the check - we don't catch those cases.